base/intfuncs.jl

```diff
@@ -149,8 +155,8 @@ lcm(a::Real, b::Real, c::Real...) = lcm(a, lcm(b, c...))
 gcd(a::T, b::T) where T<:Real = throw(MethodError(gcd, (a,b)))
 lcm(a::T, b::T) where T<:Real = throw(MethodError(lcm, (a,b)))
 
-gcd(abc::AbstractArray{<:Real}) = reduce(gcd, abc; init=zero(eltype(abc)))
-lcm(abc::AbstractArray{<:Real}) = reduce(lcm, abc; init=one(eltype(abc)))
+gcd(abc::AbstractArray{<:Real}) = reduce(gcd, abc; init=abc[1])
+lcm(abc::AbstractArray{<:Real}) = reduce(lcm, abc; init=abc[1])
```

Comment:
```suggestion
gcd(abc::AbstractArray{<:Real}) = reduce(gcd, abc)
lcm(abc::AbstractArray{<:Real}) = reduce(lcm, abc)
```

As per @barucden's suggestion in the linked issue. This prevents the OOB load and gives a better error mesage.